src/utils.js

```diff
@@ -310,4 +343,43 @@ export const convertSubfolderArray = (folderOrderArray, rawFolderContents, subfo
 export const generateImageorFilePath = (customPath, fileName) => {
   if (customPath) return encodeURIComponent(`${customPath}/${fileName}`)
   return fileName
+}
+
+const b64toBlob = (b64Data, contentType='', sliceSize=512) => {
+  const byteCharacters = atob(b64Data);
+  const byteArrays = [];
+
+  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
+    const slice = byteCharacters.slice(offset, offset + sliceSize);
+
+    const byteNumbers = new Array(slice.length);
+    for (let i = 0; i < slice.length; i++) {
+      byteNumbers[i] = slice.charCodeAt(i);
+    }
+
+    const byteArray = new Uint8Array(byteNumbers);
+    byteArrays.push(byteArray);
+  }
+
+  const blob = new Blob(byteArrays, {type: contentType});
+  return blob;
+}
+
+export async function fetchImageURL(siteName, filePath) {
+  var rawURL = `https://raw.githubusercontent.com/isomerpages/${siteName}/staging/${filePath}${filePath.endsWith('.svg') ? '?sanitize=true' : ''}`
+  const response = await fetch(rawURL);
+  //If the image is public, return the link to the raw file, otherwise make a call to the backend API to retrieve the image blob
+  if (response.ok) {
+    return rawURL;
+  } else {
+    const fileName = filePath.slice(filePath.lastIndexOf('/')+1)
+    const customPath = filePath.slice(filePath.indexOf('/')+1, filePath.lastIndexOf('/')+1).replaceAll('/','')
```

Comment:
This section could be broken up - the use of these indexes might be difficult to understand without comments or intermediate variables